src/main/java/mate/academy/security/MyUserDetailsService.java

```diff
@@ -0,0 +1,22 @@
+package mate.academy.security;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.repository.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class MyUserDetailsService implements UserDetailsService {
+    private final UserRepository repository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return repository.findByEmail(username)
+                .orElseThrow(() -> new EntityNotFoundException(
```

Comment:
you can throw UsernameNotFoundException and remove it from the signature

Reply:
Hi. Liliia, but according to the requirements at the controller level, this exception must be thrown (screen is attached)
![image](https://github.com/ystankevych/book-app/assets/105653924/a8ee3583-a2f3-40fb-bcfd-0068b4749562)

Reply:
I'll leave it for now, but look forward to your comments on this issue. If you tell me to change it, I will do it